homeassistant/components/vacuum/xiaomi_miio.py

```diff
@@ -383,3 +399,48 @@ def update(self):
             _LOGGER.error("Got OSError while fetching the state: %s", exc)
         except DeviceException as exc:
             _LOGGER.warning("Got exception while fetching the state: %s", exc)
+
+    async def async_clean_zone_start(self,
+                                     zone,
+                                     repeats: int = 1,
+                                     repeats_template: str = None,
+                                     test: bool = 0):
+        """Clean selected area for the number of repeats indicated."""
+        if repeats_template is not None:
+            _LOGGER.debug("Using template repeats")
+            repeats_template.hass = self.hass
+            repeats = repeats_template.async_render()
+            try:
+                repeats = int(repeats)
+            except ValueError:
+                repeats = 1
+                _LOGGER.debug("Failed to cast template. Set default > 1")
+        if repeats < 1:
+            repeats = 1
+        if repeats > 3:
+            repeats = 3
+        _LOGGER.debug("Start clean zone: %s repeats: %s", zone, repeats)
+        from miio import DeviceException
+        try:
+            for _zone in zone:
+                _LOGGER.debug("original zone: %s", _zone)
+                _zone.append(repeats)
+                _LOGGER.debug("zone with repeats: %s", _zone)
+        except OSError as exc:
```

Comment:
This exception should never happen. If you want to debug log the zones, it is enough to do it once outside the loop.